controllers/trainingjob/trainingjob_controller.go

```diff
@@ -185,8 +191,24 @@ func (r *Reconciler) reconcileTrainingJob(ctx reconcileRequestContext) error {
 		break
 
 	case sagemaker.TrainingJobStatusCompleted:
-		if err = r.addModelPathToStatus(ctx); err != nil {
-			return r.updateStatusAndReturnError(ctx, ReconcilingTrainingJobStatus, "", errors.Wrap(err, "Unable to add model path to status"))
+		if len(ctx.TrainingJob.Status.ModelPath) == 0 {
+			if err = r.addModelPathToStatus(ctx); err != nil {
+				return r.updateStatusAndReturnError(ctx, ReconcilingTrainingJobStatus, "", errors.Wrap(err, "Unable to add model path to status"))
+			}
+		}
+		// debug rule runs asynchronously with trainingjob. In some cases debug rule
+		// lags behind the trainingjob. Hence Training has to keep reconciling until
+		// its debug rule has completed.
+		// This breaks a bit k8s experience, since describe output will change evel
+		// if trainingjob has been completed.
+		// Its hard to populate the debug rules in `kubectl get` since not every trainifjccinlfg
+		// job will have debug rule and  column prints are not supported conditionally.
+		for _, debugRuleJob := range ctx.TrainingJob.Status.DebugRuleEvaluationStatuses {
+			if *(debugRuleJob.RuleEvaluationStatus) == string(sagemaker.RuleEvaluationStatusInProgress) || *(debugRuleJob.RuleEvaluationStatus) == string(sagemaker.RuleEvaluationStatusStopping) {
+				if err = r.addDebugRuleEvaluationStatusesToStatus(ctx); err != nil {
```

Comment:
What happens if multiple `RuleEvaluationStatus` are `InProgress` or `Stopping`? This will generate N updates for N such rules. We should fix to stop looping in that case.

Reply:
good point, I think adding `break` helps here.